### ISSUE TITLE
Web UI Testing

### ISSUE THREAD

**Issue by KipCrossing** [voteflux/voting_app#1]
It is important when developing to check the UI for both the mobile and web apps.

I came across my first challenge when hitting the [ausbills API]('https://2hqxgjv66f.execute-api.ap-southeast-2.amazonaws.com/dev/bills')

```dart
final response = await http.get(
      'https://2hqxgjv66f.execute-api.ap-southeast-2.amazonaws.com/dev/bills');
```

Works file for mobile but throw a null error for web when running `flutter run -d chrome`. A possible solution [here](https://stackoverflow.com/questions/58557173/how-to-make-http-request-work-in-flutter-web).

1) how do we get this working?
2) how to automate the setup so this works straight away with other devs?

I'm a bit out of my depth here so ill push what I've done to a different branch: *ausbillsAPI*.

**Comment by ptboyer** [comment]
As the solution link that you posted suggests it's probably a CORS issue, where are you run on mobile (Android I presume?) requests that fail CORS would pass through successfully, however running on Chrome those sam requests will actually be blocked by the browser client itself.

So you'll probably need to have the AusBills API return the correct CORS header to allow origin "*" to use the API anywhere rather than the same hostname/origin.

One workaround would be to have a small script that proxies your request/response for you but also adds the needed CORS header.

Easiest solution is to just have the AusBills API fix it on their end.

**Comment by KipCrossing** [comment]
Cool! Well, the link above won't be the actual API, that was just me testing and playing (First time writing APIs). Will keep that in mind when doing the serverless implementation.

**Comment by KipCrossing** [comment]
@XertroV **said:**

oh yeah CORS doesn't work on the API because you're not returning all the headers you need to. They aren't added to the request for you' they're allowed the be added to the request, so you still have to add them to every response (a good reason to include some abstraction layers)

particularly you need allowed origin and allowed methods, which needs OPTIONS and any other methods as used

**Comment by KipCrossing** [comment]
Not needed for beta
